docs/specification/draft/basic/primitive_authorization.mdx

```diff
@@ -0,0 +1,302 @@
+      "inputSchema": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "string",
+            "description": "The file path where the file should be created"
+          },
+          "content": {
+            "type": "string",
+            "description": "The content to write to the file"
+          }
+        },
+        "required": ["path", "content"]
+      },
+      "annotations": {
+        "title": "Create File",
+        "readOnlyHint": false,
+        "destructiveHint": true,
+        "idempotentHint": false,
+        "openWorldHint": true
+      }
+    }
+  ]
+}
+```
+
+Note that the `authorization` field is completely absent from the client response.
+
+### Authorization Evaluation Process
+
+For each primitive during discovery, the server **MUST**:
+
+1. Check if the primitive has an `authorization` component
+2. If no `authorization` component exists, apply server-wide authorization policy
+3. If `authorization` component exists, evaluate the specified policy requirements
+4. Include the primitive in the response only if authorization succeeds
+
+```mermaid
+flowchart TD
+    A[Client Request] --> B[Extract Auth Context]
+    B --> C[For Each Primitive]
+    C --> D{Has Authorization Component?}
+    D -->|No| E[Apply Server Policy]
+    D -->|Yes| F[Evaluate Primitive Policy]
+    E --> G{Authorized?}
+    F --> G
+    G -->|Yes| H[Include in Response]
+    G -->|No| I[Exclude from Response]
+    H --> J[Next Primitive]
+    I --> J
+    J --> K{More Primitives?}
+    K -->|Yes| C
+    K -->|No| L[Return Filtered Results]
+```
+
+## Runtime Authorization
+
+### Invocation-Time Checks
+
+When a client attempts to invoke a restricted primitive, the server **MUST**:
+
+1. Re-evaluate authorization policies at invocation time.
+2. Verify that the client still has the required permissions.
+3. Handle cases where additional authorization may be required.
+
+### Authorization Challenges
+
+If a primitive requires additional permissions beyond those available in the current
+authorization context, the server **MUST**:
+
+1. Respond with `HTTP 401 Unauthorized`
+2. Include a `WWW-Authenticate` header following RFC 9728 Protected Resource Metadata
+3. Provide the resource metadata URL for client authorization server discovery
+
+Example response:
+
+```
+HTTP/1.1 401 Unauthorized
+WWW-Authenticate: Bearer error="insufficient_scope",
+                  error_description="Additional scopes required: files:write workspace:modify",
+                  resource="https://mcp.example.com/.well-known/oauth-protected-resource"
+```
+
+The client **MUST** then follow the Protected Resource Metadata discovery process
+as defined in the [Authorization specification](./authorization) to obtain the
+necessary additional permissions.
+
+## Policy Integration
+
+### Policy Engine Flexibility
+
+The authorization framework is designed to be policy engine-agnostic. Servers **MAY**
```

Comment:
I agree with this goal but its not clear to me how this works. Is the `authorization`  field in the JSON example above just an opaque blob that can contain any values?